compiler/rustc_span/src/source_map.rs

```diff
@@ -167,9 +167,17 @@ struct SourceMapFiles {
     stable_id_to_source_file: UnhashMap<StableSourceFileId, Lrc<SourceFile>>,
 }
 
+/// Used to construct a `SourceMap` with `SourceMap::with_inputs`.
+pub struct SourceMapInputs(
+    pub Box<dyn FileLoader + Send + Sync>,
+    pub FilePathMapping,
+    pub SourceFileHashAlgorithm,
+);
```

Comment:
Would it be better to use a named struct?

Reply:
ok